src/modm/platform/core/rp/module.lb

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2019, Ethan Slattery
+# Copyright (c) 2021, Niklas Hauser
+# Copyright (c) 2022, Andrey Kunitsyn
+#
+# This file is part of the modm project.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+# -----------------------------------------------------------------------------
+
+import math
+
+def init(module):
+    module.name = ":platform:core"
+    module.description = FileReader("module.md")
+
+def prepare(module, options):
+    if options[":target"].identifier.platform != "rp":
+        return False
+
+    module.depends(":platform:cortex-m")
+    module.depends(":platform:clockgen")
+
+    return True
+
+
+def build(env):
+    target = env[":target"].identifier
+    env.substitutions = {"target": target}
+    env.outbasepath = "modm/src/modm/platform/core"
+    # startup helper code
+    env.template("startup_platform.cpp.in")
+
+    # delay code that must be tuned for each family
+    # (cycles per loop, setup cost in loops, max cpu frequency)
+    tuning = {
+        "rp20": (3, 4,  48), # @todo
```

Comment:
See this example for tuning, it measures against SysTick on ARMv6-M:
```suggestion
        "rp20": (3, 4), # @todo
```
https://github.com/modm-io/modm/blob/develop/examples/generic/delay/main.cpp

There's also a detailed blog post about this here:
https://blog.salkinium.com/modm-delay/